### ISSUE TITLE
[Bug]: NVDA does not read content when Carousel page changes

### ISSUE THREAD

**Issue by nikki9696** [microsoft/fluentui#34158]
### Component

Carousel

### Package version

9.61.6

### React version

18

### Environment

```shell
StackBlitz
```

### Current Behavior

Open NVDA and have it read content. Go to Carousel and find the pager button. Enter key to activate and the page changes, but the content isn't read.

### Expected Behavior

I would expect NVDA to read the content of the CarouselCard. It never seems to. If there are other attributes,  handlers, or hooks we need to use, it's not documented. (I did add the announcer)

Sample from NVDA text viewer
Preview page  frame    clickable  1 of 6  property page  Call to action  button  
Autoplay  toggle button  not pressed  
Go to prev  button  
Carousel slide 6 of 6. 
tab control
Carousel Nav Button 5  tab  selected  6 of 6
Go to next  button  
Carousel slide 1 of 6. 

### Reproduction

https://stackblitz.com/edit/uhxmpdch?file=src%2FApp.tsx

### Steps to reproduce

Open NVDA
Go to StackBlitz
Focus a button
Advance the slides
Not much is narrated

### Are you reporting an Accessibility issue?

yes

### Suggested severity

Urgent - No workaround and Products/sites are affected

### Are you willing to submit a PR to fix?

no

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by nikki9696** [comment]
It's unfortunate that we want so badly to support accessibility but the examples are not very good at showing us all the moving parts needed. Thank you!
